flow/cmd/custom_sync.go

```
@@ -0,0 +1,171 @@
+package cmd
+
+import (
+	"context"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"log/slog"
+
+	"github.com/PeerDB-io/peer-flow/generated/protos"
+	"github.com/PeerDB-io/peer-flow/peerdbenv"
+	peerflow "github.com/PeerDB-io/peer-flow/workflows"
+
+	"crypto/sha256"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const peerdbPauseGuideDocLink = "https://docs.peerdb.io/features/pause-mirror"
+
+func AuthenticateSyncRequest(ctx context.Context) error {
+	var values []string
+	var token string
+
+	md, ok := metadata.FromIncomingContext(ctx)
+	if ok {
+		values = md.Get("authorization")
+	}
+
+	if len(values) > 0 {
+		token = values[0]
+	}
+
+	peerdbPassword := peerdbenv.PeerDBPassword()
+	hash := sha256.New()
+	_, err := hash.Write([]byte(peerdbPassword))
```

nit: this never errors so we can ignore err https://pkg.go.dev/hash#Hash